### ISSUE TITLE
Rename paper.md to something with joss in the filename.

### ISSUE THREAD

**Issue by moorepants** [openjournals/joss#255]
I think it would be nice to supported a less generic `paper.md` filename, for example: `joss-v0.2.0.md` that identifies this as a JOSS related file. I added the software version number too, as it may be useful if the software is submitted multiple times for major version changes (not sure if that is possible). `paper.md` is generic enough that people may be using it for other things. You could also consider supporting it as a hidden file too: `.joss-v0.2.0.md`.

**Comment by labarba** [comment]
I've had the concern that @moorepants brings up. If authors make a major update to their software and want to submit a new paper to JOSS, with the expanded code base living on the same repo as the previous paper, what to do?

Now, once a paper is accepted in JOSS, the PDF of the paper lives in the `joss-papers` repo. So we could say that the original `paper.md` file in the software repo is no longer needed, and can be simply modified for a new submission. But somehow this feels unsatisfying.

**Comment by arfon** [comment]
So the problem here is automation: Right now with the [Whedon gem](https://github.com/openjournals/whedon) we simply look for a file named `paper.md` when we try and compile the paper with Pandoc. If we moved away from a standard filename we'd still have to keep some conventions in place (e.g. `joss-v0.2.0.md` as you suggest @moorepants).

> You could also consider supporting it as a hidden file too

IMO this is a little user-unfriendly?

**Comment by labarba** [comment]
Understood, @arfon. Even so, calling it `josspaper.md` would be a little nicer.

Also, I'm still unsure what we'll do for a second submission that is a major upgrade of existing software, already published in JOSS. Simply update the `paper.md` file?

**Comment by moorepants** [comment]
Yes, `josspaper.md` would be preferable to me too. As far as the hidden file goes, my suggestion stemmed from the fact that having to add a non-software necessary file in the source repo seemed a bit odd to me, so hidden seemed more appropriate. Publishing software to Zenodo, for example, doesn't require any files to be placed in the source repo.

**Comment by kyleniemeyer** [comment]
Something like `josspaper.md` or `joss-paper.md` sounds good. I imagine that for a new version of the software being submitted to JOSS just updating that same paper file would work fine (version control FTW!). I imagine JOSS keeps a record of each version associated with the paper.


> Publishing software to Zenodo, for example, doesn't require any files to be placed in the source repo.

@moorepants "Publishing" is perhaps a strong word for archiving something with Zenodo—a JOSS software paper includes an actual paper, so (to me at least) it isn't too onerous to ask for one extra file in the source repo. Plus, most projects already have plenty of non-software files like CONTRIBUTING.md, CODE_OF_CONDUCT.md, etc., and the JOSS paper is a brief description of the software, so perhaps not unreasonable to include in the repo anyways?

(However, if there's a simple solution to putting the paper somewhere else, I'm sure we could consider it.)

**Comment by moorepants** [comment]
Another idea here, that follows github. Can we have a `.joss` directory in the root of the repo for all the joss related items?

**Comment by moorepants** [comment]
I put my joss related files into a sub-directory in my repo named "joss-paper" and submitted the paper. It worked! So it seems you can put the paper.md in a sub-directory. So this can be close wrt to my issue. Having a sub-directory works for me.

**Comment by arfon** [comment]
> I put my joss related files into a sub-directory in my repo named "joss-paper" and submitted the paper. It worked! So it seems you can put the paper.md in a sub-directory. So this can be closed wrt to my issue. Having a sub-directory works for me.

Sorry, I didn't realize this would be acceptable to you. Is has been possible for the `paper.md` file to be anywhere in the repository since day 1 of JOSS. 😁
